python/wasmedge_sock/wasi_socket_ext.c

```diff
@@ -700,71 +701,56 @@ int getpeername(int sockfd, struct sockaddr* restrict addr,
   return 0;
 }
 
-int getsockname(int fd, struct sockaddr* restrict addr,
+int getsockname(int sockfd, struct sockaddr* restrict addr,
                 socklen_t* restrict addrlen) {
-  // addrlen contains the size of the struct pointed at by addr
-  // it needs to be returned with the actual size of the address
-
-  if (addr == NULL || addrlen == NULL) {
-    errno = EFAULT;
-    return -1;
-  }
+  WSEDEBUG("WASME| getpeername[%d]: fd:'%d' addr:'%p' size:'%d' \n", __LINE__,
+           sockfd, addr, *addrlen);
 
   uint32_t port;
-  we_address_buffer raw;
-  we_address waddr = {
-      .buf = &raw,
-      .len = sizeof(raw),
-  };
-
-  int32_t err = __import_wasip1_sock_getlocaladdr(fd, &waddr, &port);
-  if (0 != err) {
-    errno = err;
+  we_address_buffer buf;
+  we_address rsa = {.buf = &buf, .len = sizeof(buf)};
+  int32_t res = __import_wasip1_sock_getlocaladdr(sockfd, &rsa, &port);
+  if (res != 0) {
+    WSEDEBUG("WASME| bad getpeeraddr %d => %d\n", sockfd, res);
+    errno = res;
     return -1;
   }
-  socklen_t max = *addrlen;
 
-  switch (raw.family) {
+  // Intermediate output address to make truncation easier, for the cost of an
```

Comment:
That code looked familiar, took me a second to realize it was copied from `getpeername` :D 

Any  chance this can be factored into a separate function? maybe `we_address_to_sockaddr` to mirror the existing `sockaddr_to_we_address`?

Reply:
I think it’s should be needed in getpeername too, it was likely a bug.